### PR TITLE
added /cpu-config api tests

### PR DESCRIPTION
**Pull request by ShadowCurse**
Rework of #3623 in python
## Changes
added /cpu-config api tests

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by pb8o — APPROVED** [review]

**Review by mattschlebusch — APPROVED** [review]
tl;dr - No blockers from my side. 

I do think there is some overlap/repetition of this test logic in [functional/test_cpu_features_aarch64.p](https://github.com/firecracker-microvm/firecracker/blob/feature/cpu-templates/tests/integration_tests/functional/test_cpu_features_aarch64.py#L120), but we can streamline later if need be.
